### ISSUE TITLE
Carto Css Nultiparameters

### ISSUE THREAD

**Issue by lasyakoechlin** [CartoDB/mobile-sdk#439]
Is there a way to define more complex objects for nultiparameters and to query them in cartocss?

For example, I've used things like:
```
"nutiparameters": {
    "use_3d": {
      "default": false
    },
    "selected_id": {
      "default": -1
    },
    "lang": {
      "default": "en"
    },
    "fallback_lang": {
      "default": "en"
    }
}
```

And, then in carto css I can do things like:
```
polygon-fill: ([ID] = [nuti::selected_id] ? @selected_fill : @unselected_fill);
```

What I would like to do is something more like this:
```
"nutiparameters": {
    "ids": [
    ]
  }
```
Where now, in code, I could programatically update this list of integers and then in my Carto CSS I could have something like:
```
polygon-fill: ( [nuti::ids"].contains([ID]) =  ? @selected_fill : @unselected_fill);
```

I know one very ugly method could be to allow for multiple nultiparameters like:
```
"nutiparameters": {
    "id_1": -1,
    "id_2": -1,
    "id_3": -1,
    "id_4": -1
  }
```

And then do something like:

```
polygon-fill: ([ID] = [nuti::id_1] 
                          ? @selected_fill 
                          :  ([ID] = [nuti::id_2]
                              ? @selected_fill 
                              : ([ID] = [nuti::id_3]
                                   ? @selected_fill 
                                   : ([ID] = [nuti::id_4]
                                       ? @selected_fill 
                                       :  @unselected_fill))));
```

But, for obvious reasons, that's super ugly and not very easily scaled. 

I'm not sure if `OR` operators are supported, but if so, that would be slightly better... Something like:

```
polygon-fill: (([ID] = [nuti::id_1]  || [ID] = [nuti::id_2] || [ID] = [nuti::id_3] || [ID] = [nuti::id_4])
                          ? @selected_fill 
                          :  @unselected_fill);
```

Any suggestions would be greatly appreciated! Thanks.

**Comment by farfromrefug** [comment]
That would be awesome indeed. I would love to maps too. For now i store font icons as one parameter per icon. Having a mao would make it ten times easier to use and maintain

**Comment by mtehver** [comment]
There is no support for lists for nutiparameters, as vector tile features only support basic types. But I think there is a simple workaround, simply encode the list as a regular expression and use it for matching the id. Something like this:

```
polygon-fill: ("[id]" =~ [nuti::selected_ids] ? @selected_fill : @unselected_fill);
```

`nuti::selected_ids` should be encoded as follows: `(^123$)|(^1999$)`, if you want to match 123 and 1999, for example.

**Comment by lasyakoechlin** [comment]
Hi, the regex is a very interesting solution. Thanks for sharing. I tried implementing it but had some issues. In my debugging, I was able to confirm that the regex was generated successfully and I was able to use an online regex calculator to verify that, for example, in your above case `123` or `1999` would both be found successfully. However, I was not able to see polygon fill reflected accordingly. I will continue to investigate more and report back, as needed. Thanks
